src/utils/core/logger.py

```
@@ -0,0 +1,194 @@
+
+def shutdown_logging() -> None:
+    """Remove all Loguru sinks to flush queued messages. Safe to call multiple times."""
+    global _sinks_initialized, _console_sink_id, _file_sink_ids
+    # Attempt to remove file sinks; log failures at debug level with exception info
+    for sid in list(_file_sink_ids.values()):
+        try:
+            _loguru_logger.remove(sid)
+        except Exception as e:
+            try:
+                logger.debug(f"Error removing file sink id={sid}", exc_info=True)
+            except Exception:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)